src/Grids/latitude_longitude_grid.jl

```diff
@@ -239,11 +239,10 @@ function with_precomputed_metrics(grid)
                                              Δxᶠᶜᵃ, Δxᶜᶠᵃ, Δxᶠᶠᵃ, Δxᶜᶜᵃ, Δyᶠᶜᵃ, Δyᶜᶠᵃ,
                                              Azᶠᶜᵃ, Azᶜᶠᵃ, Azᶠᶠᵃ, Azᶜᶜᵃ, grid.radius)
 end
-
+    
 function validate_lat_lon_grid_args(topology, size, halo, FT, latitude, longitude, z, precompute_metrics)
     if !isnothing(topology)
-        TX, TY, TZ = validate_topology(topology)
-        Nλ, Nφ, Nz = size = validate_size(TX, TY, TZ, size)
+        TX, TY, TZ = topology
```

Comment:
this change here, why? this is what causes your trouble with Flat z coord @francispoulin

Reply:
Agreed!  @simone-silvestri made this change and I don't pretend to understand the details.  

Have worked on this for a while now and it seems clear this is a tricky problem.  I remember there were some recent discussions about lat-lon grid and wanting to allow for Flat.  Shallow water only needs flat in z but if we want to allow for flat in x and y as well, which seems reasoanble, then some care should be taken to do this well.

Maybe this part should be part of another PR?

Reply:
I am not sure about this change here. On main everything seems to work correctly 

```julia
julia> grid = LatitudeLongitudeGrid(size = (100, 10), latitude = (-10, 10), longitude = (-10, 10), topology = (Periodic, Bounded, Flat))
100×10×1 LatitudeLongitudeGrid{Float64, Periodic, Bounded, Flat} on CPU with 3×3×0 halo and with precomputed metrics
├── longitude: Periodic λ ∈ [-10.0, 10.0)    regularly spaced with Δλ=0.2
├── latitude:  Bounded  φ ∈ [-10.0, 10.0]    regularly spaced with Δφ=2.0
└── z:         Flat z

julia> model = ShallowWaterModel(; grid, gravitational_acceleration=9.0, formulation=VectorInvariantFormulation(), momentum_advection=VectorInvariant())
┌ Warning: The ShallowWaterModel is currently unvalidated, subject to change, and should not be used for scientific research without adequate validation.
└ @ Oceananigans.Models.ShallowWaterModels ~/development/Oceananigans.jl/src/Models/ShallowWaterModels/shallow_water_model.jl:129
ShallowWaterModel{typename(CPU), Float64}(time = 0 seconds, iteration = 0)
├── grid: 100×10×1 LatitudeLongitudeGrid{Float64, Periodic, Bounded, Flat} on CPU with 3×3×0 halo and with precomputed metrics
├── tracers: ()
└── coriolis: Nothing
```
we can probably revert these changes

Reply:
I should revert `latitude_longitude_grid.jl` to what we have on `main`?

I can give that a try and see if I can get this going but it seems that the lat-lon grid is a bit fussy when you declare a direction to be `Flat`.

Reply:
I reverted to the old file and that seemed to work.  Thanks!  Now will work on the validation example and see if that runs.